client/client.go

```diff
@@ -48,11 +48,12 @@ type Client interface {
 	GetPrevRegion(ctx context.Context, key []byte) (*metapb.Region, *metapb.Peer, error)
 	// GetRegionByID gets a region and its leader Peer from PD by id.
 	GetRegionByID(ctx context.Context, regionID uint64) (*metapb.Region, *metapb.Peer, error)
-	// ScanRegion gets a list of regions, starts from the region that contains key.
+	// ScanRegions gets a list of regions, starts from the region that contains key.
 	// Limit limits the maximum number of regions returned.
+	// Timeout sets context timeout for scanRegions, set to 0 will use default timeout(3s)
 	// If a region has no leader, corresponding leader will be placed by a peer
 	// with empty value (PeerID is 0).
-	ScanRegions(ctx context.Context, key, endKey []byte, limit int) ([]*metapb.Region, []*metapb.Peer, error)
+	ScanRegions(ctx context.Context, key, endKey []byte, limit int, timeout time.Duration) ([]*metapb.Region, []*metapb.Peer, error)
```

Comment:
Why doesn't use the `ctx` to set the timeout?

Reply:
[fixed](https://github.com/pingcap/pd/pull/1895/commits/35d2538c835755542a25de5c7ad9ea5292394547#diff-1cd41a2611da49ce63630d5a51bd82c5R662)